dbms/src/Storages/KVStore/MultiRaft/PrehandleSnapshot.cpp

```diff
@@ -66,24 +68,70 @@ struct ReadFromStreamResult
     RegionPtr region;
 };
 
+void PreHandlingTrace::waitForSubtaskResources(uint64_t region_id, size_t parallel, size_t parallel_subtask_limit)
+{
+    {
+        auto current = ongoing_prehandle_subtask_count.load();
+        if (current + parallel <= parallel_subtask_limit
+            && likely(ongoing_prehandle_subtask_count.compare_exchange_weak(current, current + parallel)))
+        {
+            LOG_DEBUG(
+                log,
+                "Prehandle resource meet, limit={}, current={}, region_id={}",
+                parallel_subtask_limit,
+                ongoing_prehandle_subtask_count.load(),
+                region_id);
+            return;
+        }
```

Comment:
```suggestion
        if (likely(current + parallel <= parallel_subtask_limit
            && ongoing_prehandle_subtask_count.compare_exchange_weak(current, current + parallel)))
        {
            LOG_DEBUG(
                log,
                "Prehandle resource meet, limit={} current={} parallel={} region_id={}",
                parallel_subtask_limit,
                ongoing_prehandle_subtask_count.load(),
                parallel,
                region_id);
            return;
        }
```

Reply:
Why?`current + parallel <= parallel_subtask_limit` is not `likely` if there are too many prehandle snapshot tasks from proxy.

Reply:
`likely` macro works on branching prediction to let the code generator reorganize the asm code following the if condition. I think there is no meaning to marking "half" of the if condition as `likely`

Reply:
Yes you are right, I think I should split it into two `if`s?

Reply:
Well, I think this is not the critical path for branching prediction missing, so the `likely` here does not act as an important part. So I'd like to keep the code simple.
Just one `if` with or without `likely` is OK to me.

Reply:
Makes sense